lib/rspec/matchers/built_in/captures.rb

```diff
@@ -0,0 +1,77 @@
+module RSpec
+  module Matchers
+    module BuiltIn
+      # @api private
+      # Provides the regex with captures matcher
+      class Captures < BaseMatcher
+        def initialize(*args, &blk)
+          super(*args, &blk)
+          @expected_captures = nil
+        end
+
+        # Used to specify the captures we match against
+        # @return [self]
+        def with_captures(*captures)
+          @expected_captures = captures
+          self
+        end
+
+        # @api private
+        # @return [String]
+        def description
+          return BuiltIn::Match.new(expected).description unless @expected_captures
+          "match with string #{@expected} have captures #{surface_descriptions_in(@expected_captures).inspect}"
+        end
+
+      private
+
+        def match(expected, actual)
+          return BuiltIn::Match.new(expected).matches?(actual) unless Regexp === actual
+          return BuiltIn::Match.new(expected).matches?(actual) unless @expected_captures
```

Comment:
Although minor perf wise, this is a bunch of extra instantiations of the same thing (3 across the class)